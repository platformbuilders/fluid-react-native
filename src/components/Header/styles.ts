import styled from 'styled-components/native';
import { ThemeProps } from '@platformbuilders/theme-toolkit';

import { getTheme, ifStyle } from '../../utils/helpers';
import Icon from '../Icon';

import TypographyComponent from '../Typography';

type WrapperStyleProps = {
  isIOSPlatform: boolean;
};

type ThemePrimaryStyleProps = {
  isThemePrimary: boolean;
};

const isThemePrimary = ifStyle('isThemePrimary');
const isIOSPlatform = ifStyle('isIOSPlatform');
const spacingSm = getTheme('spacing.sm');
const spacingMd = getTheme('spacing.md');
const getThemePrimaryContrastTextColor = (props: ThemeProps) =>
  isThemePrimary(
    getTheme('brand.primary.contrast')(props),
    getTheme('info.contrast')(props),
  )(props);
const getThemePrimaryContrastBackgroundColor = (props: ThemeProps) =>
  isThemePrimary(
    getTheme('brand.primary.main')(props),
    getTheme('background.z4')(props),
  )(props);

export const Wrapper = styled.View<WrapperStyleProps & ThemePrimaryStyleProps>`
  ${isIOSPlatform('overflow: hidden; padding-bottom: 5px;', '')};
`;

export const ContentWrapper = styled.View<
  WrapperStyleProps & ThemePrimaryStyleProps
>`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-horizontal: ${spacingMd}px;
  padding-vertical: ${spacingMd}px;
  height: 64px;
  background-color: ${getThemePrimaryContrastBackgroundColor};
  ${isIOSPlatform('', 'elevation: 3;')};
`;

export const TextWrapper = styled.View`
  flex: 1;
  align-items: center;
`;

export const Title = styled(TypographyComponent).attrs({
  variant: 'lg',
})<ThemePrimaryStyleProps>`
  color: ${getThemePrimaryContrastTextColor};
  font-weight: bold;
`;

export const Subtitle = styled(TypographyComponent).attrs({
  variant: 'sm',
})<ThemePrimaryStyleProps>`
  color: ${getThemePrimaryContrastTextColor};
  margin-top: ${spacingSm}px;
`;

export const StyledIcon = styled(Icon).attrs((props) => ({
  color: getThemePrimaryContrastTextColor(props),
}))<ThemePrimaryStyleProps>``;
