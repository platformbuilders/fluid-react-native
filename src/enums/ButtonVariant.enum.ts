export enum ButtonVariant {
  FILLED = 'filled',
  TINT = 'tint',
  FLAT = 'flat',
  GHOST = 'ghost',
  OUTLINE = 'outline',
}
