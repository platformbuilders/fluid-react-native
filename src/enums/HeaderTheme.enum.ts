export enum HeaderTheme {
  PRIMARY = 'PRIMARY',
  SECONDARY = 'SECONDARY',
}
