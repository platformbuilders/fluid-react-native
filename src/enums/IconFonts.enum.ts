export enum IconFonts {
  Material = 'material',
  FABrands = 'fa-brands',
  FALight = 'fa-light',
  FARegular = 'fa-regular',
  FASolid = 'fa-solid',
}
