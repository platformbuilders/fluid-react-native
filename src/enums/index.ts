export * from './InputStatus.enum';
export * from './IconFonts.enum';
export * from './HeaderTheme.enum';
export * from './TextVariant.enum';
