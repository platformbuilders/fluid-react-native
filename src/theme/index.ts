export { default as colors } from './colors';
export { default as spacing } from './spacing';
export { default as radius } from './radius';
export { default as animations } from './animations';
export { default as typography } from './typography';
