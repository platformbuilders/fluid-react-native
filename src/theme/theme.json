{
  "spacing": {
    "min": {
      "value": 1,
      "type": "spacing"
    },
    "xxs": {
      "value": 2,
      "type": "spacing"
    },
    "xs": {
      "value": 4,
      "type": "spacing"
    },
    "sm": {
      "value": 8,
      "type": "spacing"
    },
    "md": {
      "value": 16,
      "type": "spacing"
    },
    "lg": {
      "value": 24,
      "type": "spacing"
    },
    "xl": {
      "value": 32,
      "type": "spacing"
    },
    "xxl": {
      "value": 48,
      "type": "spacing"
    },
    "max": {
      "value": 64,
      "type": "spacing"
    }
  },
  "sizing": {
    "min": {
      "value": 1,
      "type": "sizing"
    },
    "xxs": {
      "value": 2,
      "type": "sizing"
    },
    "xs": {
      "value": 4,
      "type": "sizing"
    },
    "sm": {
      "value": 8,
      "type": "sizing"
    },
    "md": {
      "value": 16,
      "type": "sizing"
    },
    "lg": {
      "value": 24,
      "type": "sizing"
    },
    "xl": {
      "value": 32,
      "type": "sizing"
    },
    "xxl": {
      "value": 48,
      "type": "sizing"
    },
    "max": {
      "value": 64,
      "type": "sizing"
    }
  },
  "borderRadius": {
    "min": {
      "value": 0,
      "type": "borderRadius"
    },
    "xxs": {
      "value": 1,
      "type": "borderRadius"
    },
    "xs": {
      "value": 2,
      "type": "borderRadius"
    },
    "sm": {
      "value": 4,
      "type": "borderRadius"
    },
    "md": {
      "value": 6,
      "type": "borderRadius"
    },
    "lg": {
      "value": 8,
      "type": "borderRadius"
    },
    "xl": {
      "value": 16,
      "type": "borderRadius"
    },
    "xxl": {
      "value": 32,
      "type": "borderRadius"
    },
    "max": {
      "value": 999,
      "type": "borderRadius"
    }
  },
  "borderWidth": {
    "min": {
      "value": 0.5,
      "type": "borderWidth"
    },
    "xxs": {
      "value": 1,
      "type": "borderWidth"
    },
    "xs": {
      "value": 1.5,
      "type": "borderWidth"
    },
    "sm": {
      "value": 2,
      "type": "borderWidth"
    },
    "md": {
      "value": 3,
      "type": "borderWidth"
    },
    "lg": {
      "value": 4,
      "type": "borderWidth"
    },
    "xl": {
      "value": 6,
      "type": "borderWidth"
    },
    "xxl": {
      "value": 8,
      "type": "borderWidth"
    },
    "max": {
      "value": 12,
      "type": "borderWidth"
    }
  },
  "boxShadow": {
    "z0": {
      "value": {
        "x": 0,
        "y": 0,
        "spread": 0,
        "color": "",
        "blur": 0
      },
      "type": "boxShadow"
    },
    "z1": {
      "value": {
        "x": 0,
        "y": 1,
        "spread": 0,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 3
      },
      "type": "boxShadow"
    },
    "z2": {
      "value": {
        "x": 0,
        "y": 4,
        "spread": -1,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 6
      },
      "type": "boxShadow"
    },
    "z3": {
      "value": {
        "x": 0,
        "y": 10,
        "spread": -3,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 15
      },
      "type": "boxShadow"
    },
    "z4": {
      "value": {
        "x": 0,
        "y": 20,
        "spread": -5,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 25
      },
      "type": "boxShadow"
    },
    "z5": {
      "value": {
        "x": 0,
        "y": 25,
        "spread": -12,
        "color": "rgba(0, 0, 0, 0.2)",
        "blur": 50
      },
      "type": "boxShadow"
    }
  },
  "brand": {
    "primary": {
      "light": {
        "value": "#7684F9",
        "type": "color"
      },
      "main": {
        "value": "#4255EF",
        "type": "color"
      },
      "dark": {
        "value": "#1B288E",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "secondary": {
      "light": {
        "value": "#8BD6F8",
        "type": "color"
      },
      "main": {
        "value": "#42ADEF",
        "type": "color"
      },
      "dark": {
        "value": "#337BB7",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "tertiary": {
      "light": {
        "value": "#BA96F5",
        "type": "color"
      },
      "main": {
        "value": "#8442EF",
        "type": "color"
      },
      "dark": {
        "value": "#4A00DC",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "accent": {
      "light": {
        "value": "#FBD268",
        "type": "color"
      },
      "main": {
        "value": "#F49D19",
        "type": "color"
      },
      "dark": {
        "value": "#A1530A",
        "type": "color"
      },
      "contrast": {
        "value": "#351B02",
        "type": "color"
      }
    }
  },
  "success": {
    "light": {
      "value": "#F4FCD1",
      "type": "color"
    },
    "main": {
      "value": "#96C920",
      "type": "color"
    },
    "dark": {
      "value": "#396006",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "danger": {
    "light": {
      "value": "#FFE8D8",
      "type": "color",
      "description": "the 100 shade of a color"
    },
    "main": {
      "value": "#FF4F3F",
      "type": "color"
    },
    "dark": {
      "value": "#7A0C27",
      "type": "color",
      "description": "the 900s shade of a color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "warning": {
    "light": {
      "value": "#FFF8CE",
      "type": "color"
    },
    "main": {
      "value": "#FFCA0C",
      "type": "color"
    },
    "dark": {
      "value": "#7A5502",
      "type": "color"
    },
    "contrast": {
      "value": "#261B00",
      "type": "color"
    }
  },
  "info": {
    "light": {
      "value": "#C2ECEC",
      "type": "color"
    },
    "main": {
      "value": "#42CCD4",
      "type": "color"
    },
    "dark": {
      "value": "#186F82",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "text": {
    "main": {
      "value": "#444444",
      "type": "color"
    },
    "dark": {
      "value": "#000",
      "type": "color"
    },
    "light": {
      "value": "#999999",
      "type": "color"
    },
    "contrast": {
      "value": "#fff",
      "type": "color"
    }
  },
  "background": {
    "z0": {
      "value": "#ffffff",
      "type": "color"
    },
    "z1": {
      "value": "#fafafa",
      "type": "color"
    },
    "z2": {
      "value": "#f0f0f0",
      "type": "color"
    },
    "z3": {
      "value": "#dddddd",
      "type": "color"
    },
    "z4": {
      "value": "#d1d1d1",
      "type": "color"
    }
  },
  "opacity": {
    "disabled": {
      "value": "30%",
      "type": "opacity"
    },
    "empty": {
      "value": "50%",
      "type": "opacity"
    },
    "overlay": {
      "value": "80%",
      "type": "opacity"
    }
  },
  "fontFamilies": {
    "heading": {
      "value": "Roboto",
      "type": "fontFamilies"
    },
    "body": {
      "value": "Roboto",
      "type": "fontFamilies"
    }
  },
  "lineHeights": {
    "heading": {
      "value": "110%",
      "type": "lineHeights"
    },
    "body": {
      "value": "140%",
      "type": "lineHeights"
    }
  },
  "letterSpacing": {
    "default": {
      "value": 0,
      "type": "letterSpacing"
    },
    "bigger": {
      "value": "105%",
      "type": "letterSpacing"
    }
  },
  "paragraphSpacing": {
    "h1": {
      "value": 32,
      "type": "paragraphSpacing"
    },
    "h2": {
      "value": 26,
      "type": "paragraphSpacing"
    }
  },
  "fontWeights": {
    "headingRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "headingBold": {
      "value": "Bold",
      "type": "fontWeights"
    },
    "bodyRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "bodyBold": {
      "value": "Bold",
      "type": "fontWeights"
    }
  },
  "fontSizes": {
    "min": {
      "value": 10,
      "type": "fontSizes"
    },
    "xxs": {
      "value": 11,
      "type": "fontSizes"
    },
    "xs": {
      "value": 13,
      "type": "fontSizes"
    },
    "sm": {
      "value": 14,
      "type": "fontSizes"
    },
    "md": {
      "value": 15,
      "type": "fontSizes"
    },
    "lg": {
      "value": 17,
      "type": "fontSizes"
    },
    "xl": {
      "value": 20,
      "type": "fontSizes"
    },
    "xxl": {
      "value": 24,
      "type": "fontSizes"
    },
    "max": {
      "value": 32,
      "type": "fontSizes"
    }
  },
  "themeRadius": {
    "button": {
      "value": 3,
      "type": "borderRadius"
    },
    "card": {
      "value": 6,
      "type": "borderRadius"
    },
    "input": {
      "value": 3,
      "type": "borderRadius"
    }
  },
  "typography": {
    "H1": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "H2": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "Body": {
      "type": "typography",
      "value": {
        "fontFamily": "$fontFamilies.body",
        "fontWeight": "$fontWeights.bodyRegular",
        "lineHeight": "$lineHeights.heading",
        "fontSize": "$fontSizes.body",
        "paragraphSpacing": "$paragraphSpacing.h2"
      }
    }
  }
}
