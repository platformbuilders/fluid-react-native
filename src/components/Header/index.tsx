import React, { Children, FC, ReactChild } from 'react';

import { isIOS } from '@platformbuilders/helpers/native';

import { HeaderTheme, IconFonts } from '../../enums';

import {
  ContentWrapper,
  StyledIcon,
  Subtitle,
  TextWrapper,
  Title,
  Wrapper,
} from './styles';

interface BaseProps {
  theme?: string;
  subtitle?: string;
  showLeftIcon?: boolean;
  showRightIcon?: boolean;
  leftIconName?: string;
  rightIconName?: string;
  accessibilityLeftIcon?: string;
  accessibilityRightIcon?: string;
  onPressLeftIcon?: () => void;
  onPressRightIcon?: () => void;
}

interface PropsWithTitleRequired extends BaseProps {
  title: string;
  children?: ReactChild | ReactChild[];
}

interface PropsWithChildrenRequired extends BaseProps {
  title?: string;
  children: ReactChild | ReactChild[];
}

type HeaderProps = PropsWithTitleRequired | PropsWithChildrenRequired;

export const DEFAULT_LEFT_ICON_NAME = 'arrow-left';
export const DEFAULT_RIGHT_ICON_NAME = 'times';

const Header: FC<HeaderProps> = ({
  children,
  title,
  theme = HeaderTheme.PRIMARY,
  subtitle = '',
  accessibilityLeftIcon = '',
  accessibilityRightIcon = '',
  showLeftIcon = true,
  showRightIcon = true,
  leftIconName = DEFAULT_LEFT_ICON_NAME,
  rightIconName = DEFAULT_RIGHT_ICON_NAME,
  onPressLeftIcon,
  onPressRightIcon,
}) => {
  const hasChildren = Children.count(children) > 0;
  const isIOSPlatform = isIOS();
  const isThemePrimary = theme === HeaderTheme.PRIMARY;

  return (
    <Wrapper isThemePrimary={isThemePrimary} isIOSPlatform={isIOSPlatform}>
      <ContentWrapper
        isIOSPlatform={isIOSPlatform}
        isThemePrimary={isThemePrimary}
      >
        {showLeftIcon && leftIconName?.length > 0 && (
          <StyledIcon
            isThemePrimary={isThemePrimary}
            accessibility={accessibilityLeftIcon || leftIconName}
            type={IconFonts.FontAwesome}
            onPress={onPressLeftIcon}
            name={leftIconName}
          />
        )}
        {hasChildren ? (
          children
        ) : (
          <TextWrapper>
            <Title isThemePrimary={isThemePrimary}>{title}</Title>
            {subtitle?.length > 0 && (
              <Subtitle isThemePrimary={isThemePrimary}>{subtitle}</Subtitle>
            )}
          </TextWrapper>
        )}
        {showRightIcon && rightIconName?.length > 0 && (
          <StyledIcon
            isThemePrimary={isThemePrimary}
            accessibility={accessibilityRightIcon || rightIconName}
            type={IconFonts.FontAwesome}
            onPress={onPressRightIcon}
            name={rightIconName}
          />
        )}
      </ContentWrapper>
    </Wrapper>
  );
};

export default Header;
