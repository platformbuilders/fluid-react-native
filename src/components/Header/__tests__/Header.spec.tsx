import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import { fireEvent, render } from '@testing-library/react-native';

import Header, { DEFAULT_LEFT_ICON_NAME, DEFAULT_RIGHT_ICON_NAME } from '..';
import { HeaderTheme } from '../../../enums';
import theme from '../../../theme';

const onPressLeftIcon = jest.fn();
const onPressRightIcon = jest.fn();

describe('<Header />', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render Header', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Header
          title="Cadastro"
          subtitle="O seu saldo é de R$2.500,00"
          theme={HeaderTheme.PRIMARY}
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should trigger onPressLeftIcon function when left icon is pressed', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Header
          title="Cadastro"
          subtitle="O seu saldo é de R$2.500,00"
          theme={HeaderTheme.PRIMARY}
          onPressLeftIcon={onPressLeftIcon}
          onPressRightIcon={onPressRightIcon}
        />
      </ThemeProvider>,
    );

    const component = getByTestId(DEFAULT_LEFT_ICON_NAME);

    fireEvent.press(component);

    expect(onPressLeftIcon).toHaveBeenCalledTimes(1);
    expect(onPressRightIcon).not.toHaveBeenCalled();
  });

  it('should trigger onPressRightIcon function when right icon is pressed', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Header
          title="Cadastro"
          subtitle="O seu saldo é de R$2.500,00"
          theme={HeaderTheme.PRIMARY}
          onPressLeftIcon={onPressRightIcon}
          onPressRightIcon={onPressRightIcon}
        />
      </ThemeProvider>,
    );

    const component = getByTestId(DEFAULT_RIGHT_ICON_NAME);

    fireEvent.press(component);

    expect(onPressRightIcon).toHaveBeenCalledTimes(1);
    expect(onPressLeftIcon).not.toHaveBeenCalled();
  });
});
