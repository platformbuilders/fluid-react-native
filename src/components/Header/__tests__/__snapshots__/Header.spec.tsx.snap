// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header 1`] = `
<View
  isIOSPlatform={true}
  isThemePrimary={true}
  style={
    Array [
      Object {
        "overflow": "hidden",
        "paddingBottom": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "elevation": 3,
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      isIOSPlatform={true}
      isThemePrimary={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#C6B09E",
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "space-between",
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          },
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={Object {}}
      >
        <View
          accessibilityLabel="arrow-left"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="arrow-left"
        >
          <View>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <Text
          accessibilityLabel="Cadastro"
          isThemePrimary={true}
          lineHeightVariant="min"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 17,
                "lineHeight": 17,
              },
              Object {
                "color": "#ffffff",
                "fontWeight": "bold",
              },
            ]
          }
          testID="Cadastro"
          variant="lg"
        >
          Cadastro
        </Text>
        <Text
          accessibilityLabel="O seu saldo é de R$2.500,00"
          isThemePrimary={true}
          lineHeightVariant="min"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
              },
              Object {
                "color": "#ffffff",
                "marginTop": 12,
              },
            ]
          }
          testID="O seu saldo é de R$2.500,00"
          variant="sm"
        >
          O seu saldo é de R$2.500,00
        </Text>
      </View>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={Object {}}
      >
        <View
          accessibilityLabel="times"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="times"
        >
          <View>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
